tests/rptest/services/producer_swarm.py

```diff
@@ -135,16 +168,19 @@ def is_metrics_available(self, node):
         except:
             return False
 
-    def is_alive(self, node):
-        result = node.account.ssh_output(
+    def is_alive(self):
+        result = self._node.account.ssh_output(
             f"bash /opt/remote/control/alive.sh {self.EXE}")
         result = result.decode("utf-8")
         return "YES" in result
 
     def wait_node(self, node, timeout_sec=600):
-        self._redpanda.wait_until(lambda: not self.is_alive(node),
-                                  timeout_sec=timeout_sec,
-                                  backoff_sec=5)
+        try:
```

Comment:
This function used to throw TimeoutError, now it does not (return bool instead).  Is this gonna be fine for all callers?

Nit: add `->bool` return type?
General wish: we really ought to add static type checking to leverage all our nice annotations.

Reply:
> Nit: add ->bool return type?

Added (type checker already infers the correct type almost always but doesn't hurt to add it).

> General wish: we really ought to add static type checking to leverage all our nice annotations.

I am adding this soon, I'll put you on the PR.

Reply:
Return type added in https://github.com/redpanda-data/redpanda/compare/5d9ad58e47691dbc16c82a9e1b5b413cab4307e0..837694d18838e6329dab858c39c2e930e831ccfc